### PR TITLE
#1426 GroupCreationModal reachable even when no group is created

### LINKED ISSUE #1426: GroupCreationModal reachable even when no group is created
### Problem

The proper guards for joining a group are missing and little validation is done. Visiting <http://localhost:3000/app/?modal=GroupCreationModal&step=GroupWelcome> results in a success screen with empty group information (name), even when this screen was not reached as part of a group joining process.

### Solution

The state should be validated properly to ensure that this screen is reachable only upon joining a group successfully as part of the group creation flow.

### PR DESCRIPTION
**Pull request by SebinSong**
closes #1426

### CODE REVIEW

File: frontend/views/components/modal/Modal.vue [hunk 1]
```diff
@@ -154,6 +158,11 @@ export default ({
     },
     setModalQueries (componentName, queries) {
       this.queries[componentName] = queries
+    },
+    isPreauthGuarded (toModal) {
```
Comment (SebinSong) [inline frontend/views/components/modal/Modal.vue]: While investigating this issue, noticed that we are able to open `GroupCreationModal.vue` even when the user is not logged in yet, when this modal is only meant to be used for authenticated users. So decided to go ahead and work on adding a guard logic in `$route` watcher of `Modal.vue` here. (Pls let me know if any of you think there is a better way of achieving this..!)

File: frontend/views/containers/access/SignupForm.vue [hunk 2]
```diff
@@ -71,7 +70,6 @@ export default ({
     validationsDebouncedMixins
   ],
   components: {
-    ModalTemplate,
```
Comment (SebinSong) [inline frontend/views/containers/access/SignupForm.vue:74]: spotted an unused local component here, so removed it even though it's not related to this bug.

File: frontend/views/components/modal/Modal.vue (hunk 1 above)
Comment (snowteamer) [reply frontend/views/components/modal/Modal.vue]: Can't this be achieved by adding a guard in `~/frontend/controller/router.js`?

Comment (SebinSong) [reply frontend/views/components/modal/Modal.vue]: @snowteamer agreed the `beforeEach()` guard in `router.js` is more suitable place to handle this logic so moved it there in the following commit.

File: frontend/controller/router.js [hunk 3]
```diff
@@ -168,6 +168,15 @@ const router: any = new Router({
 })
 
 router.beforeEach((to, from, next) => {
+  if (to.query.modal &&
+    !['SignupModal', 'LoginModal'].includes(to.query.modal) &&
+    !store.state.loggedIn) {
+    // if modal is queried and,
+    // the requested modal is only meant to be used post authentication and,
+    // the user is not logged in now, discard the navigation.
```
Comment (taoeffect) [inline frontend/controller/router.js:176]: Excellent comment 👍

**Review by taoeffect — APPROVED** [review]
LGTM!

**Pull request merged**